### ISSUE TITLE
Hi i have a problem!

### ISSUE THREAD

**Issue by Flipp3rix** [openhardwaremonitor/openhardwaremonitor#834]
In VB using the open hardware monitor lib i can't acces to core clock speed...
i'm using 
`If sensor.SensorType = SensorType.Clock Then
                        Label7.Text = (sensor.Name)
                        Label8.Text = sensor.Value
                    End If`

but the program show me memory clock...how can i get the core clock? Thanks in Advance!

**Comment by DarthDestroyer** [comment]
1. You need a Computer! Make it Public, load it at startup...

`   Dim cp As New Computer()
    cp.Open()
    cp.HDDEnabled = True
    cp.RAMEnabled = True
    cp.GPUEnabled = True
    cp.MainboardEnabled = True
    cp.CPUEnabled = True`

Enable / Disable Hardware that you need / not need.

2. Foreach through cp.Hardware

`For Each hardwareitem As OpenHardwareMonitor.Hardware.IHardware In cp.Hardware
hardwareitem.Update()
If hardwareitem.HardwareType = HardwareType.CPU Then
For Each hardwareitemsensor As OpenHardwareMonitor.Hardware.ISensor In hardwareitem.Sensors
If hardwareitemsensor.SensorType = SensorType.Clock Then 
Label7.Text = hardwareitemsensor.Name
Label8.Text = hardwareitemsensor.Value.ToString()
End If
Next
End if
Next
`

3. After using, close your Computer ;)

**Comment by Flipp3rix** [comment]
Ok, but when i use this code
```
For Each hw In cp.Hardware

            If hw.HardwareType = HardwareType.GpuAti Then
                hw.Update()
For Each sensor In hw.Sensors
                    If sensor.SensorType = SensorType.Clock Then
                        Label7.Text = (sensor.Name)
                        Label8.Text = sensor.Value
                    End If
Next
End If
Next
```
Label7and8 show me only memory clock of my GPUAti and not core clock...help me and thanks in Advance!

**Comment by DarthDestroyer** [comment]
of course... any clock is in your for each, and the memory clock is the last in the row, so... this will be displayed! you override your "Label.Text" everytime with the for each!

You could handle this with:
`If sensor.name.contains('Memory') Then
LabelMemoryClock.Text = sensor.Value.ToString()
End If`

If you want to know, what is in the for each, i would do it so:
(Instead of your 2 labels)
`Console.WriteLine(sensor.name & sensor.value)`
with this, you will get all Names and the corresponding values!

**Comment by Flipp3rix** [comment]
Thanks!!!!! It worked!! Thanks so much!!

**Comment by DarthDestroyer** [comment]
Could you post the Solution? That the Answer will be available for others?

If your Problem is solved, could you close this Issue?

Regards
